lms/djangoapps/courseware/views/views.py

```diff
@@ -225,7 +226,10 @@ def jump_to(_request, course_id, location):
     except InvalidKeyError:
         raise Http404(u"Invalid course_key or usage_key")
     try:
-        redirect_url = get_redirect_url(course_key, usage_key)
+        if usage_key.block_type == 'course' and waffle.flag_is_active(request, 'unified_course_view'):
+            redirect_url = reverse('edx.course_experience.course_home', args=[unicode(course_key)])
+        else:
+            redirect_url = get_redirect_url(course_key, usage_key)
```

Comment:
It seems like this logic belongs in `get_redirect_url`.  What do you think of adding a temporary third parameter of `unified_course_view` to the call, which is more likely to be deleted than temporarily adding `request`.  Let me know what you think.

I also wonder whether that call should just be in this view if this is the only caller.

While you are in the code, can you clean-up this unused import of the call?:
https://github.com/edx/edx-platform/blob/5c26acc1157de996ca2744bb61f9fb439fa83b00/lms/djangoapps/verify_student/views.py#L32-L32